### ISSUE TITLE
bug(source): null value for `sequence` type fields in `datagen` source

### ISSUE THREAD

**Issue by eemario** [risingwavelabs/risingwave#7460]
### Describe the bug

According to https://www.risingwave.dev/docs/current/create-source-datagen/, users can specify `start` and `end` for `sequence` type fields. It seems that `null` will be generated when the number exceeds this range.

### To Reproduce

```
CREATE MATERIALIZED SOURCE s1 (id int, name varchar) 
WITH (
     connector = 'datagen',
     fields.id.kind = 'sequence',
     fields.id.start = '1',
     fields.name.kind = 'random',
     fields.name.length = '16',
     fields.name.seed = '3',
     datagen.rows.per.second = '1000'
 ) ROW FORMAT JSON;

SELECT * FROM s1 WHERE id is null;
```

```
 id |       name       
----+------------------
    | bJDXYmDVsP7VpWl8
    | cSHsYFi77HVY3gZK
    | gDVliYCbWJF0adeF
    | uyT7ZZGFg9lZllUp
    | Cplh5MDSGo5Tivef
...
```

### Expected behavior

Maybe stop generating data when `sequence` type numbers are out of bound.

**Comment by lmatz** [comment]
probably add an additional field to specify the total number of messages
as there can be multiple fields specifying the `start` and `end`,

an explicit parameter seems better, compared to an implicit `min(...)`

I got no strong opinion on this

**Comment by waruto210** [comment]
I suggest adding an explicit parameter, such as `total_msg`, and then datagen will generate `min(total_msg, (end-start))` rows of data

**Comment by waruto210** [comment]
fixed bt #7755
